### ISSUE TITLE
The format of the .pt of --data in '''Image denoising using PFGM++'''

### ISSUE THREAD

**Issue by Chelxen** [dennishein/cpfgmpp_PCCT_denoising#1]
Hi, when I reproduce this part , I get an error report.

In this section '''Image denoising using PFGM++''', you mentioned dicom files, but after entering the --data parameter it actually reads .pt files, I didn't find the formatting requirements for this conditional data in your Readme, so I'm getting an error here and I can't solve the problem. 

I wonder if you can describe how this .pt file should be obtained? How should his data format be?

I hope you can help me out with this, I would appreciate it!

Best

**Comment by dennishein** [comment]
Thanks for raising this issue! 

In this setting --data points to the data you want to run inference on. The data will be a torch tensor of size [n_slices, 2, res, res]. The values will be in Hounsfield units (HU). There are two channels because I have concatenated the LDCT (observed, first dimension) and NDCT (label, second dimension) images. If you don't have paired data, just repeat the noisy images you wish to process. Whilst not strictly necessary, I usually also clamp the HU values using the min/max in the Mayo data (-1024/3071). 

You can find some helpful functions specifically for the Mayo data here: https://github.com/SSinyu/RED-CNN/blob/master/prep.py

Hope this helps!

**Comment by Chelxen** [comment]
> Thanks for raising this issue!
> 
> In this setting --data points to the data you want to run inference on. The data will be a torch tensor of size [n_slices, 2, res, res]. The values will be in Hounsfield units (HU). There are two channels because I have concatenated the LDCT (observed, first dimension) and NDCT (label, second dimension) images. If you don't have paired data, just repeat the noisy images you wish to process. Whilst not strictly necessary, I usually also clamp the HU values using the min/max in the Mayo data (-1024/3071).
> 
> You can find some helpful functions specifically for the Mayo data here: https://github.com/SSinyu/RED-CNN/blob/master/prep.py
> 
> Hope this helps!

Thanks for your reply! It works.

**Comment by Chelxen** [comment]
HI, I noticed that the training dataset feed into 'pt_to_np_mayo_1mm' are also a pt with size [n_slices, 2, res, res],so I guess it has the same channel setting (LD/ND) right? But when the data send into the 'dataset_tool_cond.py',it should have some label setting here to make the data paired as the supervise learning said in the paper, can you please tell me how to deal with the label when processing these training data? I felt so confused here.

Hope to receive your reply! Best

**Comment by dennishein** [comment]
Right! You can actually just ignore the label setting, as it will not be used. Using pt_to_np_mayo_1mm.ipynb, pointing to an original tensor with size [n_slices, 2, res, res], for channel setting LD/ND, and then running 
```.bash
python dataset_tool_cond.py --source=./datasets_unzipped/train_mayo_1_alt/ \
    --dest=datasets/mayo_1mm_alt-512x512.zip
```
will put the data on correct format for the rest of the code. 

Hope this helps!

**Comment by Chelxen** [comment]
Thanks for your reply!

I just do not understand how the supervised learning work, because I tried to train my own dataset with the same operations as yours :original tensor [n_slices, 2, res, res] and obtain [res,res,2] as the correct format for the input and I also set a [n_slices, 2, res, res] conditional data during sampling ,but it seems that the model do not work as expected ,the loss is lower and lower but the generated results looked the same as the LD input.

I guess it may be the problem that the model can not figure out the paired data?
